Automattic/wp-calypso
# Add: Recovery phone validation flow

## Summary
This PR adds the validation flow for account recovery phones. e.g.
![demo-validate-phone](https://cloud.githubusercontent.com/assets/1842898/21631290/79c3f282-d274-11e6-90e3-ef4f0bb88591.gif)

Note that it is only available for users without binding 2fa sms numbers, simply because we blocks those users from editing recovery phone numbers at the moment. After deploying this PR, I will open another PR for removing the block. Just wanted to make the whole process gradual to play safe.

## Test Plan
Please make sure you have deployed D3859 to your test environment. Otherwise, you won't receive the validation code. All the following steps assume you are logged in as a user without binding a 2FA sms number.

Case 1: Add a recovery phone number

* Make sure the recovery phone field is empty. If not, remove it.
* Enter a valid phone number, see the validation notice prompt, and then you should receive a validation code shortly.
* Enter the code, click validate, and see it successful.
* Refresh the page. There shouldn't be any validation notice, which means the number is validated.

Case 2: Update a recovery phone number

* Make sure the recovery phone field is not empty.
* Update the number, see the validation notice prompt, and then you should receive a validation code shortly.
* Enter the code, click validate, and see it successful.
* Refresh the page. There shouldn't be any validation notice, which means the number is validated.

Case 3: Request a new validation code

* Add or update the phone field, so that the number is not validated.
* Refresh the page, and you should find a "Resend" button underneath the validation notice.
* Hit "Resend", and then you should receive a new validation code shortly.
* Use the code to validate the number in the same way as above.
